src/main/java/somethingrandom/dataaccess/local/FileUserDataAccessObject.java

```diff
@@ -0,0 +1,131 @@
+package somethingrandom.dataaccess.local;
+
+import org.json.JSONObject;
+import org.json.JSONWriter;
+import somethingrandom.entity.*;
+import somethingrandom.usecase.DataAccessException;
+
+import java.io.*;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.*;
+
+public class FileUserDataAccessObject {
+    private final File dataFile;
+    private final Map<UUID, Map<String, String>> items;
+    private final ItemFactory factory;
+
+    public FileUserDataAccessObject(String dataFilePath, ItemFactory factory) throws DataAccessException {
+        dataFile = new File(dataFilePath);
+        items = new HashMap<>();
+        this.factory = factory;
+        if (dataFile.length() == 0) {
+            save();
+        } else {
+            load();
+        }
+    }
+
+    private void load(){
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(dataFile));
+            JSONObject data = new JSONObject(reader.readLine());
+            reader.close();
+            for (String key: data.keySet()) {
+                if (!data.keySet().contains(key)) {
+                    Map<String, String> itemData = new HashMap<>();
+                    JSONObject item = data.getJSONObject(key);
+                    for (String itemKey : item.keySet()) {
+                        if (!itemKey.equals("empty")) {
+                            itemData.put(itemKey, item.getString(itemKey));
+                        }
+                    }
+                    items.put(UUID.fromString(key), itemData);
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Can't you have a `throws DataAccessException` and `throw new DataAccessException(e)`?